server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -235,6 +235,15 @@ public abstract class BlobStoreRepository extends AbstractLifecycleComponent imp
         Setting.Property.NodeScope
     );
 
+    /**
+     * Setting to set batch size of stale blobs to be deleted.
+     */
+    public static final Setting<Integer> MAX_SHARD_BLOB_DELETE_BATCH_SIZE = Setting.intSetting(
+        "max_shard_blob_delete_batch_size",
+        1000,
```

Comment:
@nknize : Thanks for looking into the PR. I kept default value to be 1000 because most of cloud repository (e.g: S3 and GCP) allow batch deletion of upto 1000 documents at a time. But again I am open for suggestion on default value. 
Also thanks for suggesting to compute batch value dynamically. Let me think how we can compute it dynamically and raise another PR for it. 

For the documentation part,  `./gradlew missingJavadoc` is `BUILD SUCCESSFUL` on my dev machine. 
Can you please guide where/how shall I update doc for this new setting?

Reply:
> Can you please guide where/how shall I update doc for this new setting?

Some more details about why a user might want to consider a larger or smaller batch size would be helpful. For example, does the S3 implementation break if I set this to 1001? Does performance get much worse if I set this to 10?

Reply:
Updated details of `MAX_SHARD_BLOB_DELETE_BATCH_SIZE`